tests/system/data_sources/test_teradata.py

```diff
@@ -477,6 +478,62 @@ def test_row_validation_binary_pk_to_bigquery():
     binary_key_assertions(df)
 
 
+@mock.patch(
+    "data_validation.state_manager.StateManager.get_connection_config",
+    new=mock_get_connection_config,
+)
+def test_row_validation_string_pk_to_bigquery():
+    """Teradata to BigQuery dvt_string_id row validation.
+    This is testing string primary key join columns.
+    Includes random row filter test.
+    """
+    parser = cli_tools.configure_arg_parser()
+    args = parser.parse_args(
+        [
+            "validate",
+            "row",
+            "-sc=td-conn",
+            "-tc=bq-conn",
+            "-tbls=udf.dvt_string_id=pso_data_validator.dvt_string_id",
+            "--primary-keys=id",
+            "--hash=id,other_data",
+            "--use-random-row",
+            "--random-row-batch-size=5",
+        ]
+    )
+    df = run_test_from_cli_args(args)
+    id_type_test_assertions(df)
+
+
+@mock.patch(
+    "data_validation.state_manager.StateManager.get_connection_config",
+    new=mock_get_connection_config,
+)
+def test_row_validation_char_pk_to_bigquery():
+    """Teradata to BigQuery dvt_char_id row validation.
+    This is testing CHAR primary key join columns.
+    Includes random row filter test.
+    """
+    parser = cli_tools.configure_arg_parser()
+    args = parser.parse_args(
+        [
+            "validate",
+            "row",
+            "-sc=td-conn",
+            "-tc=bq-conn",
+            "-tbls=udf.dvt_char_id=pso_data_validator.dvt_char_id",
+            "--primary-keys=id",
+            "--hash=id,other_data",
+            "--use-random-row",
+            "--random-row-batch-size=5",
```

Comment:
I did add support for trimming the PKs within random row when the `-tsp` flag is present. This should only be used when the destination is VARCHAR and source is CHAR. 

Random rows are applied at the inner-most level so adding doing random row validation with BQ padded columns is not possible. For this reason, I think we should remove the random row from this test.